orbit/pkg/luks/luks_linux.go

```diff
@@ -0,0 +1,264 @@
+//go:build linux
+
+package luks
+
+import (
+	"context"
+	"crypto/rand"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"math/big"
+	"os/exec"
+	"regexp"
+	"time"
+
+	"github.com/fleetdm/fleet/v4/orbit/pkg/dialog"
+	"github.com/fleetdm/fleet/v4/orbit/pkg/lvm"
+	"github.com/fleetdm/fleet/v4/server/fleet"
+	"github.com/rs/zerolog/log"
+	"github.com/siderolabs/go-blockdevice/v2/encryption"
+	luksdevice "github.com/siderolabs/go-blockdevice/v2/encryption/luks"
+)
+
+const (
+	entryDialogTitle     = "Enter disk encryption passphrase"
+	entryDialogText      = "Passphrase:"
+	retryEntryDialogText = "Passphrase incorrect. Please try again."
+	infoFailedTitle      = "Encryption key escrow"
+	infoFailedText       = "Failed to escrow key. Please try again later."
+	infoSuccessTitle     = "Encryption key escrow"
+	infoSuccessText      = "Key escrowed successfully."
+	maxKeySlots          = 8
+)
+
+var ErrKeySlotFull = regexp.MustCompile(`Key slot \d+ is full`)
+
+func (lr *LuksRunner) Run(oc *fleet.OrbitConfig) error {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	if !oc.Notifications.RunDiskEncryptionEscrow {
+		return nil
+	}
+
+	devicePath, err := lvm.FindRootDisk()
+	if err != nil {
+		return fmt.Errorf("Failed to find LUKS Root Partition: %w", err)
+	}
+
+	var response LuksResponse
+	key, keyslot, err := lr.getEscrowKey(ctx, devicePath)
+	if err != nil {
+		response.Err = err.Error()
+	}
+
+	response.Passphrase = string(key)
+	response.KeySlot = keyslot
+
+	if keyslot != nil {
+		salt, err := getSaltforKeySlot(ctx, devicePath, *keyslot)
+		if err != nil {
+			return fmt.Errorf("Failed to get salt for key slot: %w", err)
+		}
+		response.Salt = salt
+	}
+
+	if err := lr.escrower.SendLinuxKeyEscrowResponse(response); err != nil {
+		if err := lr.infoPrompt(ctx, infoFailedTitle, infoFailedText); err != nil {
+			log.Debug().Err(err).Msg("failed to show failed escrow key dialog")
+		}
+		return fmt.Errorf("escrower escrowKey err: %w", err)
+	}
+
+	if response.Err != "" {
+		if err := lr.infoPrompt(ctx, infoFailedTitle, response.Err); err != nil {
+			log.Debug().Err(err).Msg("failed to show failed escrow key dialog")
+		}
+		return fmt.Errorf("error getting linux escrow key: %s", response.Err)
+	}
+
+	// Show success dialog
+	if err := lr.infoPrompt(ctx, infoSuccessTitle, infoSuccessText); err != nil {
+		log.Debug().Err(err).Msg("failed to show success escrow key dialog")
+	}
+
+	return nil
+}
+
+func (lr *LuksRunner) getEscrowKey(ctx context.Context, devicePath string) ([]byte, *uint, error) {
+	device := luksdevice.New(luksdevice.AESXTSPlain64Cipher)
```

Comment:
Let's add a comment that `AES-XTS` cipher seems to be the default (I've verified on my KUbuntu and Fedora VMs).